--- obj_list.c	2018-03-17 11:08:52.798506584 -0600
+++ obj_list.c.new	2018-03-17 11:09:32.791841132 -0600
@@ -9,11 +9,12 @@
 
 /* This function is called after an object's memory has been allocated. */
 static void object_init(obj_t * obj) {
-  memset(obj, 0, sizeof(obj_t));
+  obj_init(obj);
 }
 
 /* This function is called before an object's memory is freed. */
 static void object_clear(obj_t * obj) {
+  obj_clear(obj);
 }
 
 
