--- obj_queue.c	2018-03-16 22:16:29.378601303 -0600
+++ obj_queue.c	2018-03-16 22:18:53.468605677 -0600
@@ -10,11 +10,12 @@
 
 /* This function is called after an object's memory has been allocated. */
 static void object_init(obj_t * obj) {
-  memset(obj, 0, sizeof(obj_t));
+  obj_init(obj);
 }
 
 /* This function is called before an object's memory is freed. */
 static void object_clear(obj_t * obj) {
+  obj_clear(obj);
 }
 
 
