--- src/map/int_obj_map.c	2018-03-16 21:09:50.358479882 -0600
+++ src/map/int_obj_map.c	2018-03-16 21:11:49.125150156 -0600
@@ -26,11 +26,12 @@
 
 /* This function is called after an object's memory has been allocated. */
 static void object_init(obj_t * obj) {
-  memset(obj, 0, sizeof(obj_t));
+  obj_init(obj);
 }
 
 /* This function is called before an object's memory is freed. */
 static void object_clear(obj_t * obj) {
+  obj_clear(obj);
 }
 
 
