--- imap_obj.c	2018-03-09 21:16:34.141367761 -0700
+++ imap_obj.fix.c	2018-03-09 21:16:09.761367021 -0700
@@ -63,13 +63,13 @@
 /* TODO: Initialize a value, if applicable. This function is called
  * when entries created or overwritten. */
 static void init_value(obj_t * value) {
-  memset(value, 0, sizeof(obj_t));
+  obj_init(value);
 }
 
 /* TODO: Cleanup a value, if applicable. This function is called
  * when entries are erased or overwritten. */
 static void deinit_value(obj_t * value) {
-  memset(value, 0, sizeof(obj_t));
+  obj_clear(value);
 }
 
 /*** general functionaility ***/
